### PR TITLE
Sharing: Improve sharing via email to be spam-proof

### LINKED ISSUE #448: Lock down the 'Email' sharing option to decrease utility to spammers.
https://github.com/Automattic/jetpack/blob/master/modules/sharedaddy/sharing-sources.php#L196-L324

It's possible to stuff too much data into the sender name.  Let's give the email sharing module a once over and add some sane limits.

[Update from @designsimply 4-Apr-2018]

Confirming this still appears to be an issue based on recent reports in the following summary:

* 2014-07-31 - WP Engine disables email sharing. (https://github.com/Automattic/jetpack/issues/448#issuecomment-50784872)
* 2014-09-23 - issue closed with: character limit, honeypot, uses Akismet (if enabled). (#1053)
* 2015-01-31 - WP Engine re-enables email sharing. (https://github.com/Automattic/jetpack/issues/448#issuecomment-72352961)
* 2016-11-04 - issue reopened. (https://github.com/Automattic/jetpack/issues/448#issuecomment-258566916)
* 2017-02-01 - 34sp.com host reports the problem. (3251872-t)
* 2017-06-08 - 1&1 host reports the problem. (3251872-t)
* 2017-06-26 - disabled email sharing as a default. (#7329)
* 2017-07-04 - latest public user report I could find via links above. (<a href="https://wordpress.org/support/topic/jetpack-sharing-email-can-be-abused-for-spam/page/3/#post-9283709">wporg forums</a>)
* 2017-11-01 - latest private user report noted on this issue. (767682-zen)

What to do with new issues reported where Akismet is confirmed enabled:

<blockquote>It would be more effective to post these tickets to the internal Akismet P2 for review.</blockquote>

Source: https://github.com/Automattic/jetpack/issues/448#issuecomment-294254795

### PR DESCRIPTION
**Pull request by blobaugh**
Resolves #448

### CODE REVIEW

**Comment by jeherve** [comment]
@enejb I don't see any of those changes in the current version of Jetpack, and that pull request was never merged. How did you end up fixing this issue? I couldn't find any related code that would add the enhancements you mentioned above.

**Comment by samhotchkiss** [comment]
I'm moving this to 3.5-- it seems like a much more involved fix than we should tackle on a point release

**Comment by jeherve** [comment]
@enejb The extra field is displayed to readers as well, whether you're logged in or not:

screen shot 2015-04-01 at 7 04 59 pm
screen shot 2015-04-01 at 7 05 18 pm

Could you take a look?

**Comment by enejb** [comment]
The code looks good and makes sense to me. I also tested this and it works as expected.

**Comment by jeherve** [comment]
Everything seems to work for me now that you've updated the branch. :+1:

File: modules/sharedaddy/sharing.css [hunk 1]
```diff
@@ -655,6 +655,7 @@ div.sharedaddy.sharedaddy-dark #sharing_email {
 	max-width: none;
 	padding: 1px 3px;
 }
+#jetpack-source_f_name{ display: none!important; position: absolute !important; left: -9000px; }
```
Comment (georgestephanis) [inline modules/sharedaddy/sharing.css]: Why are we compressing the css on this line in this version of the file, when none of the surrounding lines are?

**Comment by samhotchkiss** [comment]
looks like this needs rebasing

**Comment by jeherve** [comment]
cc @enejb

**Comment by enejb** [comment]
@jeherve can you test this branch again? 
@samhotchkiss I rebased it but if anyone one does any more changed to the styles it will need to be rebased again.

**Comment by jeherve** [comment]
@enejb Is it normal for me to get the following in WordPress' debug.log when using the button?

```
[17-Apr-2015 05:56:39] Array
(
    [akismet_debug] => Array
        (
            [akismet_url] => https://xxx.rest.akismet.com/1.1/comment-check
            [http_args] => Array
                (
                    [body] => blog=http://xxx&permalink=xxx&comment_type=share&comment_author=Jeremy H&comment_author_email=xxx&comment_content=Jeremy H (xxx) thinks you may be interested in the following post:

Test post with one image
xxx
&user_agent=Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.118 Safari/537.36&user_ip=xxx
                    [headers] => Array
                        (
                            [Content-Type] => application/x-www-form-urlencoded; charset=UTF-8
                            [Host] => xxx.rest.akismet.com
                            [User-Agent] => WordPress/4.2-RC1-32138 | Akismet/3.1.1
                        )

                    [httpversion] => 1.0
                    [timeout] => 15
                )

            [response] => Array
                (
                    [headers] => Array
                        (
                            [server] => nginx
                            [date] => Fri, 17 Apr 2015 05:56:39 GMT
                            [content-type] => text/plain; charset=utf-8
                            [content-length] => 5
                            [connection] => close
                            [x-akismet-server] => 10.8.86.112
                            [x-akismet-guid] => xxx
                        )

                    [body] => false
                    [response] => Array
                        (
                            [code] => 200
                            [message] => OK
                        )

                    [cookies] => Array
                        (
                        )

                    [filename] => 
                )

        )

)
```

Other than that, everything seems to work, whether Akismet is activated or not.

**Comment by enejb** [comment]
@jeherve no it is not normal, thanks for reporting it. 
The reason why it it showing up is because because you have WP_DEBUG_LOG constant set as true in your wp-config.php and also the Akismet plugin enabled. 

This has nothing do to with this PR or Jetpack. See https://plugins.trac.wordpress.org/browser/akismet/trunk/class.akismet.php#L1024
that produces the log.

**Pull request merged**